src/app-layer-ftp.c

```diff
@@ -619,13 +623,12 @@ static AppLayerResult FTPParseRequest(Flow *f, void *ftp_state, AppLayerParserSt
         SCReturnStruct(APP_LAYER_ERROR);
     }
 
-    state->input = input;
-    state->input_len = input_len;
-    /* toserver stream */
-    state->direction = 0;
+    FtpInput tmpstate;
```

Comment:
I find this name to be confusing. Can this be `ftpi` like in other places?

Additionally, you can init this as `FtpInput ftpi = { .input = input, .input_len = input_len };` which I find nice and tidy.